backend/services/office_hours/ticket.py

```diff
@@ -0,0 +1,660 @@
+    OfficeHoursTicket,
+    OfficeHoursTicketDraft,
+    OfficeHoursTicketPartial,
+)
+from ...models.office_hours.ticket_details import OfficeHoursTicketDetails
+from ...models.user import User
+
+from ...services.exceptions import ResourceNotFoundException
+
+from ..permission import PermissionService
+
+
+__authors__ = ["Sadie Amato", "Bailey DeSouza", "Meghan Sun"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class OfficeHoursTicketService:
+    """Service that performs all of the actions on the `OfficeHoursTicket` table"""
+
+    def __init__(
+        self,
+        session: Session = Depends(db_session),
+        permission_svc: PermissionService = Depends(),
+        oh_event_svc: OfficeHoursEventService = Depends(),
+    ):
+        """Initializes the database session."""
+        self._session = session
+        self._permission_svc = permission_svc
+        self._oh_event_svc = oh_event_svc
+
+    def create(
+        self, subject: User, oh_ticket: OfficeHoursTicketDraft
+    ) -> OfficeHoursTicketDetails:
+        """
+        Creates a new office hours ticket.
+
+        Args:
+            subject (User): A valid User model representing the currently logged-in user.
+            oh_ticket (OfficeHoursTicketDraft): OfficeHoursTicketDraft object to add to the table.
+
+        Returns:
+            OfficeHoursTicketDetails: The newly created OfficeHoursTicket object.
+
+        Raises:
+            PermissionError: If the logged-in user is not a section member student.
+
+        """
+        # PERMISSIONS
+
+        # Fetch Office Hours Section - Needed To Determine if User Membership
+        oh_section: OfficeHoursSection = self._get_office_hours_section_by_oh_event_id(
+            oh_ticket.oh_event.id
+        )
+
+        # Check If Current Users and Creators Are Section Members and thus have permission to create a ticket.
+        section_member_entities: list[SectionMemberEntity] = []
+
+        # Case: Current User
+        current_user_section_member_entity: SectionMemberEntity = (
+            self._check_user_section_membership(subject.id, oh_section.id)
+        )
+
+        # Only allow students to create tickets
+        if current_user_section_member_entity.member_role != RosterRole.STUDENT:
+            raise PermissionError(
+                "Only Section Member Students Are Allowed to Create Tickets."
+            )
+
+        section_member_entities.append(current_user_section_member_entity)
+
+        # Case: Remaining Creator of Ticket If Any
+        for creator in oh_ticket.creators:
+            if creator.id != subject.id:
+                section_member_entity = self._check_user_section_membership(
+                    creator.id, oh_section.id
+                )
+                section_member_entities.append(section_member_entity)
+
+        oh_event = self._session.get(
+            OfficeHoursEventEntity, oh_ticket.oh_event.id
+        ).to_model()
```

Comment:
I am wondering if there is a way to complete this logic with one query to the `section_member` table for the students in a section, and then checking if all of the students are in the section - querying is not really an expensive operation, but it might help improve readability here and prevent bugs in the future!

Reply:
You bring up a good point that the list of section member entities could be gathered from one query! I changed it so a query is made to get all section member entities given a ticket creator list of ids. Then, raise a Permission Error if not all creator ticket ids had a corresponding entity. This definitely makes it easier than having to add the entity to a list every time!

To make it readable, I also moved this permission handling logic into a helper function called `_check_ticket_creators_permissions()` (ln 594- 651). It's similar to the individual user permission checks, and I hope this better encapsulates the permission checking the `create()` function. Let me know thoughts on these changes!

Reply:
Nice, that looks good @meghansun322 !